src/shared/interfaces.ts

```diff
@@ -53,6 +53,7 @@ export interface WritableWorkspaceConfig {
 	flutterVersionFile?: string;
 	useLsp?: boolean;
 	useVmForTests?: boolean;
+	flutterBazel?: boolean;
```

Comment:
I think it may be better to name this based on what it does rather than what triggered it - for example `enableFlutterMode` or `forceFlutterMode` or similar. This would be a little more generic and might be usable for other workspace types (for example it might allow simplifying some of the other pubspec checks into the more specific workspace functions).